File: Algorithms/538. Convert BST to Greater Tree/golang/solution.recursion.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func convertBST(root *TreeNode) *TreeNode {
	total := 0
	var convert func(*TreeNode)
	convert = func(node *TreeNode) {
		if node != nil {
			convert(node.Right)
			total += node.Val
			node.Val = total
			convert(node.Left)
		}
	}
	convert(root)
	return root
}

File: Algorithms/538. Convert BST to Greater Tree/javascript/solution.recursion.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var convertBST = function (root) {
    let total = 0;
    convert(root);

    function convert(root) {
        if (!root) return;
        convert(root.right);
        total += root.val;
        root.val = total;
        convert(root.left);
    }
    return root;
};
